pipeline-model-api/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ast/ModelASTPipelineDef.java

```diff
@@ -118,7 +118,7 @@ public static String toIndentedGroovy(@Nonnull String orig) {
     @Override
     public void removeSourceLocation() {
         super.removeSourceLocation();
-        removeSourceLocationsFrom(stages, libraries, postBuild, environment, tools, options, parameters, triggers);
+        removeSourceLocationsFrom(agent, stages, libraries, postBuild, environment, tools, options, parameters, triggers);
```

Comment:
We never removed source locations from `agent` previously. I'm not sure if it mattered, but I noticed  it because some test started failing when I started storing the whole Pipeline in `ExecutionModelAction` instead of just the stages.

Reply:
Oops, yes, it does matter.  This is another reason to do this.  Cool.